### PR TITLE
netstat: Add -e argument and fix multiple bugs

### PR DESCRIPTION
**Pull request by fdellwing**
This PR adds information about the user owning the process to our netstat output.

It also fixes multiple bugs that are the result of upstream changes in JsonObject.

### CODE REVIEW

File: Userland/Utilities/netstat.cpp [hunk 1]
```diff
@@ -292,13 +292,13 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
                 }
             }
 
-            auto peer_port = if_object.get_deprecated_string("peer_port"sv).value_or({});
+            String peer_port = TRY(String::number(if_object.get_u32("peer_port"sv).value_or({})));
```
Comment (trflynn89) [inline Userland/Utilities/netstat.cpp:295]: This can be kept as `auto`.

**Comment by trflynn89** [comment]
Commit message should also explain why the feature was broken

File: Userland/Utilities/netstat.cpp (hunk 1 above)
Comment (fdellwing) [reply Userland/Utilities/netstat.cpp]: c&p error, fixed

**Comment by nico** [comment]
Is the conversion to String peripheral, or is that crucial for the fix? If the former, could it be in a separate commit? (I'm guessing just the get_u32 is needed for the actual fix?)

**Comment by fdellwing** [comment]
> Is the conversion to String peripheral, or is that crucial for the fix? If the former, could it be in a separate commit? (I'm guessing just the get_u32 is needed for the actual fix?)

Just using `get_u32` is not enough, as we write a StringView to the same variable further down.

File: Userland/Utilities/netstat.cpp [hunk 2]
```diff
@@ -316,6 +331,9 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
                 columns[peer_address_column].buffer = get_formatted_address(peer_address, peer_port);
             if (state_column != -1)
                 columns[state_column].buffer = "-";
+            if (flag_extend && user_column != -1) {
+                columns[user_column].buffer = get_formatted_user(origin_uid);
+            }
```
Comment (GMTA) [inline Userland/Utilities/netstat.cpp:336]: You can leave out these braces for single-line if bodies

File: Userland/Utilities/netstat.cpp [hunk 3]
```diff
@@ -132,6 +136,13 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return DeprecatedString::formatted("{}/{}", pid, program.value());
     };
 
+    auto get_formatted_user = [&](i32 uid) {
+        if (uid == -1)
+            return DeprecatedString("-");
+
+        return DeprecatedString::number(uid);
+    };
```
Comment (GMTA) [inline Userland/Utilities/netstat.cpp:144]: Suggestion: make these `String`s instead, and use `.to_deprecated_string()` closer to the source of the deprecated usage :^)

**Review by GMTA — COMMENTED** [review]
Seems to work well! Some nitpicks.

File: Userland/Utilities/netstat.cpp (hunk 2 above)
Comment (fdellwing) [reply Userland/Utilities/netstat.cpp:336]: Yep, there where multiple lines at some point. Forgot to drop them.

File: Userland/Utilities/netstat.cpp (hunk 3 above)
Comment (fdellwing) [reply Userland/Utilities/netstat.cpp:144]: I'm really not sure if this variant is any better..

Comment (fdellwing) [reply Userland/Utilities/netstat.cpp:145]: This should be good now.

**Review by GMTA — APPROVED** [review]

**Pull request merged**